From 88ac887c165bffe8b03cc81a74d089abc4c02cd1 Mon Sep 17 00:00:00 2001
From: Robin Gong <yibin.gong@nxp.com>
Date: Tue, 23 Jul 2019 23:42:53 +0800
Subject: [PATCH] MLK-22302-2: dmaengine: fsl-edma-v3: fix build warning with
 CONFIG_PM_SLEEP=n

Fix build waring with CONFIG_PM_SLEEP=n.

Signed-off-by: Robin Gong <yibin.gong@nxp.com>
Reviewed-by: Andy Duan <fugang.duan@nxp.com>
(cherry picked from commit dfe2a755209615f9592ed937957b2efdc3b6d6c0)
---
 drivers/dma/fsl-edma-v3.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

--- a/drivers/dma/fsl-edma-v3.c
+++ b/drivers/dma/fsl-edma-v3.c
@@ -1040,6 +1040,7 @@ static int fsl_edma3_remove(struct platf
 	return 0;
 }
 
+#ifdef CONFIG_PM_SLEEP
 static int fsl_edma3_suspend_late(struct device *dev)
 {
 	struct fsl_edma3_engine *fsl_edma = dev_get_drvdata(dev);
@@ -1097,10 +1098,11 @@ static int fsl_edma3_resume_early(struct
 
 	return 0;
 }
+#endif
 
 static const struct dev_pm_ops fsl_edma3_pm_ops = {
-	.suspend_late   = fsl_edma3_suspend_late,
-	.resume_early   = fsl_edma3_resume_early,
+	SET_LATE_SYSTEM_SLEEP_PM_OPS(fsl_edma3_suspend_late,
+				     fsl_edma3_resume_early)
 };
 
 static const struct of_device_id fsl_edma3_dt_ids[] = {
